scripts/nlp_language_modeling/prepare_packed_ft_dataset.py

```diff
@@ -109,8 +123,26 @@ def tokenize_dataset(cfg: 'DictConfig'):
         special_tokens=data_cfg.get('chat_prompt_tokens', None),
         is_test=True,
     )
-
-    return np.array([dataset[i] for i in range(len(dataset))])
+    max_seq_length = dataset.max_seq_length
+    pad_id = dataset.tokenizer.eos_id
+    pad_seq_length_to_mult = dataset.pad_seq_length_to_mult
+    dataset = np.array([dataset[i] for i in range(len(dataset))])
+    if cp_size > 1:
+      def pre_pad_dataset(data, max_length, pad_id):
+        '''
+        pad each individual data point to the length of max_length
+        '''
+        for key,val in data.items():
+          if key in {'input_ids', 'context_ids'}:
+            val = val + [pad_id] * (max_length - len(val) + 1) # add 1 for cp
+            data[key]=val
+        return
+      ceil_to_nearest = lambda n, m : (n + m - 1) // m * m
+      for data in dataset:
+        max_length = min(max_seq_length, ceil_to_nearest(len(data['input_ids']), pad_seq_length_to_mult))
+        assert max_length <= max_seq_length
```

Comment:
You have `min` in 142, so this assert seems redundant?

Reply:
Removed the assert in the new commit